File: src/test/java/application/browser/BattlefieldHelper.java
package application.browser;

import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;

import com.matag.player.PlayerType;

public class BattlefieldHelper extends AbstractCardContainerHelper {
  public static final String FIRST_LINE = "first-line";
  public static final String SECOND_LINE = "second-line";
  public static final String COMBAT_LINE = "combat-line";

  private final PlayerType playerType;
  private final String lineType;

  BattlefieldHelper(MatagBrowser matagBrowser, PlayerType playerType, String lineType) {
    super(matagBrowser);
    this.playerType = playerType;
    this.lineType = lineType;
  }

  @Override
  protected WebElement containerElement() {
    return playerTypeContainer().findElement(By.className(lineType));
  }

  private WebElement playerTypeContainer() {
    if (playerType == PlayerType.PLAYER) {
      return matagBrowser.findElement(By.id("player-battlefield"));
    } else {
      return matagBrowser.findElement(By.id("opponent-battlefield"));
    }
  }
}

File: src/test/kotlin/application/browser/BattlefieldHelper.kt
package application.browser

import com.matag.player.PlayerType
import org.openqa.selenium.By
import org.openqa.selenium.WebElement

class BattlefieldHelper internal constructor(
    matagBrowser: MatagBrowser?,
    private val playerType: PlayerType?,
    private val lineType: String
) : AbstractCardContainerHelper(matagBrowser) {
    override fun containerElement(): WebElement {
        return playerTypeContainer()!!.findElement(By.className(lineType))
    }

    private fun playerTypeContainer(): WebElement? {
        if (playerType == PlayerType.PLAYER) {
            return matagBrowser.findElement(By.id("player-battlefield"))
        } else {
            return matagBrowser.findElement(By.id("opponent-battlefield"))
        }
    }

    companion object {
        const val FIRST_LINE: String = "first-line"
        const val SECOND_LINE: String = "second-line"
        const val COMBAT_LINE: String = "combat-line"
    }
}
